src/compiler/tsc.ts

```diff
@@ -46,9 +41,25 @@ namespace ts {
         return s;
     }
 
+    function getOptionsForHelp(commandLine: ParsedCommandLine) {
+        // Sort our options by their names, (e.g. "--noImplicitAny" comes before "--watch")
+        return !!commandLine.options.all ?
+            sort(optionDeclarations, (a, b) => compareStringsCaseInsensitive(a.name, b.name)) :
+            filter(optionDeclarations.slice(), v => !!v.showInSimplifiedHelpView);
+    }
+
     export function executeCommandLine(args: string[]): void {
+        if ((args[0].toLowerCase() === "--build") || (args[0].toLowerCase() === "-b")) {

```

Comment:
I'm going to reiterate what I've said before so it's written down: _I really dislike this_. This flag disables all other normal flags, must be first, and completely changes what the executable _does_ - these all indicate that it should be a _separate executable_. We can still ship it, use common code, and everything, but it should by no means be executed by `tsc --build` when it functionally shares nothing of its interface with `tsc`. A second entrypoint, `tsbuild` would make much more sense from a functionality perspective, plus it'd make the feature more googleable (and remove the awkward comments in the `CompilerOptions` interface about how "this flag should never be set"). Plus, then `tsbuild` wouldn't be forced to use `tsc`'s command line option model, and could use a much more pleasant command-based syntax (since `tsbuild` is a much more multimodal tool than `tsc`, as an orchestrator), like `tsbuild`, `tsbuild compile`, `tsbuild dryrun`, `tsbuild watch`, and so on.